packages/core/src/lib/sign-in-experience.ts

```diff
@@ -24,8 +25,33 @@ export const validateTermsOfUse = (termsOfUse: TermsOfUse) => {
 
 const isEnabled = (state: SignInMethodState) => state !== SignInMethodState.disabled;
 
+const assertNonemptySocialConnectorIds: (value?: string[]) => asserts value is string[] = (
+  socialSignInConnectorIds
+) => {
+  assertThat(
+    socialSignInConnectorIds && socialSignInConnectorIds.length > 0,
+    'sign_in_experiences.empty_social_connectors'
+  );
+};
+
+const assertEnabledSocialConnectorIds = (
+  socialSignInConnectorIds: string[],
+  enabledConnectorInstances: ConnectorInstance[]
+) => {
+  const enabledSocialConnectorIds = new Set(
+    enabledConnectorInstances
+      .filter((instance) => instance.metadata.type === ConnectorType.Social)
+      .map((instance) => instance.connector.id)
+  );
+  assertThat(
+    socialSignInConnectorIds.every((id) => enabledSocialConnectorIds.has(id)),
+    'sign_in_experiences.invalid_social_connectors'
+  );
+};
```

Comment:
nit: could be a general one, `assertConnectorsEnabled` which regardless of the connector type, for whatever id provided, check the existence and the enable status.